Type 'DeepNullable<K>' does not satisfy the constraint 'WeakKey'

I am using:
```
NodeJS v21.7.1
typescript@5.4.2
jest-mock-extended@3.0.5
```


When I build project by command ``tsc`` I have these errors from **ts-essentials**, that is the dependency of **jest-mock-extended**

I understand that errors inside **ts-essentials**, but it effect to **jest-mock-extended**.
**jest-mock-extended** uses old version **ts-essentials** -  ^7.0.3 (latest is 9.4.1)

I created also the issue https://github.com/ts-essentials/ts-essentials/issues/381

Do you have any ideas to fix it?

```
node_modules/ts-essentials/dist/types.d.ts:67:13 - error TS2344: Type 'DeepNullable<K>' does not satisfy the constraint 'WeakKey'.
  Type 'symbol | object | (Builtin & object) | Map<unknown, unknown> | WeakMap<symbol | object | null, unknown> | Set<unknown> | WeakSet<symbol | object | null> | unknown[] | Promise<unknown> | null' is not assignable to type 'WeakKey'.
    Type 'null' is not assignable to type 'WeakKey'.

67   ? WeakMap<DeepNullable<K>, DeepNullable<V>>
               ~~~~~~~~~~~~~~~

node_modules/ts-essentials/dist/types.d.ts:71:13 - error TS2344: Type 'DeepNullable<U>' does not satisfy the constraint 'WeakKey'.
  Type 'symbol | object | (Builtin & object) | Map<unknown, unknown> | WeakMap<symbol | object | null, unknown> | Set<unknown> | WeakSet<symbol | object | null> | unknown[] | Promise<unknown> | null' is not assignable to type 'WeakKey'.

71   ? WeakSet<DeepNullable<U>>
               ~~~~~~~~~~~~~~~

node_modules/ts-essentials/dist/types.d.ts:91:13 - error TS2344: Type 'DeepUndefinable<K>' does not satisfy the constraint 'WeakKey'.
  Type 'symbol | object | (Builtin & object) | Map<unknown, unknown> | WeakMap<symbol | object | undefined, unknown> | Set<unknown> | WeakSet<symbol | object | undefined> | unknown[] | Promise<...> | undefined' is not assignable to type 'WeakKey'.
    Type 'undefined' is not assignable to type 'WeakKey'.

91   ? WeakMap<DeepUndefinable<K>, DeepUndefinable<V>>
               ~~~~~~~~~~~~~~~~~~

node_modules/ts-essentials/dist/types.d.ts:95:13 - error TS2344: Type 'DeepUndefinable<U>' does not satisfy the constraint 'WeakKey'.
  Type 'symbol | object | (Builtin & object) | Map<unknown, unknown> | WeakMap<symbol | object | undefined, unknown> | Set<unknown> | WeakSet<symbol | object | undefined> | unknown[] | Promise<...> | undefined' is not assignable to type 'WeakKey'.

95   ? WeakSet<DeepUndefinable<U>>
```

<img width="1415" alt="image" src="https://github.com/marchaos/jest-mock-extended/assets/127202468/2ae53ddc-d79f-4dbb-b267-70870adde98b">

--- a/package.json
+++ b/package.json
@@ -20,7 +20,7 @@
     "coveralls": "jest --coverage && cat ./coverage/lcov.info | coveralls"
   },
   "dependencies": {
-    "ts-essentials": "^7.0.3"
+    "ts-essentials": "^9.4.2"
   },
   "devDependencies": {
     "@types/jest": "^27.5.0",

